app/forms/personal_details_form.rb

```diff
@@ -0,0 +1,34 @@
+class PersonalDetailsForm
+  include ActiveModel::Model
+
+  attr_accessor :fullname, :email, :licence_id
+
+  def initialize(fullname: nil, email: nil, licence_id: nil)
+    @fullname = fullname
+    @email = email
+    @licence_id = licence_id
+  end
+
+  validates :fullname,
+    presence: {
+      message: "You must provide your full name"
+    }
+
+  validates :email,
+    presence: {
+      message: "You must provide your email address"
+    },
+    format: {
+      with: /.+@.+\..+/,
+      message: "Your email address must contain the @ symbol and end in a domain name (e.g. roger@example.com)"
+    }
+
+  validates :licence_id,
+    presence: {
+      message: "You must provide your Licence ID"
+    },
+    format: {
+      with: /.{8,}/,
```

Comment:
Maybe this could use the built in length validator instead of this regex so that it's easier to approach

`validates :licence_id, length: { minimum: 8 }`

Reply:
yes, thank you -- much clearer!